applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1123,12 +1057,80 @@ static void disconnect(const void *subscriber_id, uint8_t report_id)
 
 	struct report_data *rd = rs->linked_rd;
 
+	rs->linked_rd = NULL;
+
 	if (rd->linked_rs == rs) {
-		rd->linked_rs = NULL;
 		clear_report_data(rd);
+		rd->linked_rs = find_next_report_state(rd);
+		__ASSERT_NO_MSG(rd->linked_rs != rs);
+
+		if (rd->linked_rs) {
+			/* Refresh state of the newly routed subscriber. */
+			report_send(rd, true, true);
+		}
 	}
+}
 
-	rs->linked_rd = NULL;
+static void connect_subscriber(const void *subscriber_id, bool is_usb, uint8_t report_max)
```

Comment:
why you move these functions? it's harder to review :(

Reply:
move them back

Reply:
both functions should be ok btw

Reply:
I wanted to move the functions to keep functionalities related to subscriptions close to each other. hid_state is quite big module so actually this change may be helpful (only drawback is that is makes review a bit harder). Maybe we could leave it like that?

Reply:
let's leave it but please avoid moving long functions in the future

Reply:
Ok, sorry for inconvenience